### DISCUSSION TITLE
Demo

### DISCUSSION THREAD

**Discussion by jbcaillau** [control-toolbox/control-toolbox.github.io#43] [Ideas]
@ocots To be changed into an issue once the repo exists 🤞🏾

# "Journée contrôle optimal et applications" outline
## FRUMAM, June 9, 2023

@jbcaillau (20')
1. rappel contexte ct: control-toolbox (bocop + hampath, interconnexion python - *cf.* [galerie](https://ct.gitlabpages.inria.fr/gallery/notebooks.html), bonus : bocop v3 / web app)
1.  `basic` use case : fonctionnel + abstrait, direct solve)
1. `goddard` use case : abstrait, direct solve

@ocots (20')
1. `goddard` use case (suite) : indirect solve
1. `CTProblems` : vers un benchmark en contrôle optimal 
1. architecture [control-toolbox.org](https://control-toolbox.org)

**Comment by jbcaillau** [comment]
@ocots PS. I have made some minor changes in [goddard](https://github.com/control-toolbox/OptimalControl.jl/blob/revise/examples/goddard.jl), plus I suppose that we can use the `F0 ⋅ g` notation for the Lie operation (vector against function) in there:
```julia
g(x) = vmax - constraint(ocp, :eq2)(x, -1) # -1 for tf, not used
ub(x) = -Lie(F0, g)(x) / Lie(F1, g)(x)
μ(x, p) = H01(x, p) / Lie(F1, g)(x)
```

**Reply by ocots** [reply]
No it is not possible to use the dot notation since in this case we cannot tell the time and variable dependences. Another possibility would be to authorize this notation only for autonomous and fixed problems.

**Reply by jbcaillau** [reply]
WTF!?

**Comment by ocots** [comment]
Finally, I have created a repo `frunam`: https://github.com/control-toolbox/frunam

**Reply by jbcaillau** [reply]
👍🏽🙏🏽 just renamed it to FRUMAM 😅
https://github.com/control-toolbox/frumam

**Reply by jbcaillau** [reply]
... and everything works with `julia 1.9`. all set!
